package/boot/arm-trusted-firmware-mvebu/Makefile

```diff
@@ -81,32 +122,42 @@ define Build/Prepare
 	$(eval $(call Download,mv-ddr-marvell))
 	$(eval $(call Download,gcc-linaro))
 
-	rm -rf $(PKG_BUILD_DIR)
-	$(TAR) -C $(BUILD_DIR) -xf $(DL_DIR)/$(PKG_SOURCE)
+	$(call Build/Prepare/Default,)
 
 	mkdir -p $(STAGING_DIR_IMAGE)
 	$(TAR) -C $(STAGING_DIR_IMAGE) -xf $(DL_DIR)/$(A3700_UTILS_SOURCE)
 	$(TAR) -C $(STAGING_DIR_IMAGE) -xf $(DL_DIR)/$(MV_DDR_SOURCE)
 	$(TAR) -C $(STAGING_DIR_IMAGE) -xf $(DL_DIR)/$(LINARO_SOURCE)
 endef
 
-TARGET_CFLAGS = ""
+export GCC_HONOUR_COPTS=s
 
-MAKE_VARS= \
-	CROSS_COMPILE="$(TARGET_CROSS)"
+TARGET_CFLAGS = ""
 
-MAKE_FLAGS = \
-	CROSS_CM3=$(STAGING_DIR_IMAGE)/$(LINARO_NAME)-$(LINARO_RELEASE).$(LINARO_VERSION)/bin/arm-linux-gnueabi- \
-	BL33=$(STAGING_DIR_IMAGE)/$(UBOOT)-u-boot.bin \
-	MV_DDR_PATH=$(STAGING_DIR_IMAGE)/$(MV_DDR_NAME) \
-	WTP=$(STAGING_DIR_IMAGE)/$(A3700_UTILS_NAME) \
-	CLOCKSPRESET=$(CLOCKSPRESET) \
-	PLAT=$(PLAT) \
-	all \
-	fip
+define Build/Compile
+	$(eval $(Package/arm-trusted-firmware-mvebu-$(BUILD_VARIANT))) \
+	$(MAKE) $(PKG_JOBS) -C $(PKG_BUILD_DIR) \
+		CROSS_COMPILE=$(TARGET_CROSS) \
+		OPENSSL_DIR=$(STAGING_DIR_HOST) \
+		CROSS_CM3=$(STAGING_DIR_IMAGE)/$(LINARO_NAME)-$(LINARO_RELEASE).$(LINARO_VERSION)/bin/arm-linux-gnueabi- \
+		BL33=$(STAGING_DIR_IMAGE)/$(UBOOT)-u-boot.bin \
+		MV_DDR_PATH=$(STAGING_DIR_IMAGE)/$(MV_DDR_NAME) \
+		WTP=$(STAGING_DIR_IMAGE)/$(A3700_UTILS_NAME) \
+		DDR_TOPOLOGY=$(DDR_TOPOLOGY) \
+		CLOCKSPRESET=$(CLOCKSPRESET) \
+		PLAT=$(PLAT) \
+		all \
+		fip
+endef
```

Comment:
Some time ago I did this: http://ix.io/2vGK. Maybe it can give some inspiration how to handle all variants.

Reply:
The goal is not to ship a ton of variants where < 10% is stable. The goal is to ship stable mainline builds for boards which actually sold.

Reply:
This was not showed to build all possible variants (especially the clockpreset) , just to keep defines limited (`define Package/arm-trusted-firmware-mvebu-espressobin-<some-variant>`). You can ignore this comment, it was just to show different approach for generating similar output.

Reply:
Oh okay, sorry, indeed I misunderstood then!